12-watchtower-API.md

```diff
@@ -88,28 +86,32 @@ The requesting node:
 
 #### Rationale
 
-The client is always the one in charge of establishing the connection. His `wt_init` message informs the server of what type of service he is requesting. The server should agree with the client if the implement the same methods. Otherwise the connection should be failed.
+The client is always the one in charge of establishing the connection. A client that receives a `wt_init` from a random node must assume that the node tries to use him as a server, therefore fail. 
+
+The client's `wt_init` message informs the server of what type of service he is requesting. The server should agree with the client if the implement the same methods. Otherwise the connection should be failed.
 
 QoS is an optional field. Including the field in the `wt_init` message signals that the sender is requiring that specific QoS. As for accountability, it aims for giving non-repudiable proof of the agreement to both the client and the server.
 
-The transport protocol to be used is purposely ommited. Piggybacking on top of the Lightning transport protocol as well other approaches such as interfaces over HTTP can be used to establish the connection.
+The transport protocol to be used is purposely omitted. Piggybacking on top of the Lightning transport protocol as well other approaches such as interfaces over HTTP can be used to establish the connection.
+
+[FIXME: The connection establishment and wt_init can be replaced by a node discovery algorithm that lets the server announce its policy]
 
-## Sending / receiving appointments
+## Sending and receiving appointments
 
 Once both client and server have agreed on common modes of operation, the client can start sending appointments to the server.
 
-		+-------+                                +-------+
-		|   A   |--(1)---    appointment    ---->|   B   |
-		|       |<-(2)---    accept/reject  -----|       |
-		+-------+                                +-------+
+		+-------+                                    +-------+
+		|   A   |--(1)---      appointment      ---->|   B   |
+		|       |<-(2)---   accepted/rejected   -----|       |
+		+-------+                                    +-------+
 		
 		- where node A is 'client' and node B is 'server'
 
 ### The `appointment` message
```

Comment:
What's None? I don't know what you're talking about.

Start allow to book appointments in advance instead of forcing the tower to start Watching at the specific time the appointment is requested. We could argue that since the Tower already had holds the data start_time should be the request time (which I'm fine with).

So I feel the options should be either remove it all-together or leave it

Reply:
if "NONE" QoS is selected, then it makes sense to have: 

data:
[16*byte :locator]
[u64 :end_block]
[varsize:encrypted_blob]
[u16:cipher]

the customers signature could be optional (perhaps for some API-key etc, so you still have 10/100 jobs for a watchtower / help them track it). 

We could also change it to authentication-type and then you can select from macaroon, unblinded token, customer signature.

Reply:
Again, QoS is optional, it makes no sense to have a mandatory field that can be set to None in a network protocol. If it can be none it should be optional, and the lack of it implies none.

Reply:
ok but that isnt the part of this comment right? we are just talking about what to send up if NONE is set or the field is left out?

Reply:
what was the outcome for this?